### DISCUSSION TITLE
i2c

### DISCUSSION THREAD

**Discussion by wangqingpen** [edgexfoundry/community#193] [Ideas]
How to control devices under I2C under EdExfoundry

**Comment by cloudxxx8** [comment]
please describe your idea with more details

**Reply by jiekechoo** [reply]
@wangqingpen I think, the I2C sensor/device doesn’t support in EdgeX, you should integrated by a MCU device with I2C bus, and provide modbus RTU protocol.

**Reply by wangqingpen** [reply]
edgex-sdk also can't use the i2C protocol?

**Reply by cloudxxx8** [reply]
if i2C is a device protocol, it needs a new device service implementation based on device-sdk C or golang.
EdgeX doesn't provide it right now, and welcome to anyone's contribution.

**Reply by jiekechoo** [reply]
@wangqingpen As @cloudxxx8 mentioned, there is a `device-sdk-c` or `device-sdk-go` that you can refer, then implement your Device Driver to support I2C protocol. The easest way is to put I2C sensor in a Smart Device.
And contribution is welcome. 😄
